bootstrap.sh

```diff
@@ -64,11 +63,11 @@ export PYTHONPATH=$(pwd)/external:$(pwd)/external/usr/lib/python$pyver/site-pack
 CMD $python -m pip install --no-cache-dir -q --upgrade pip --target=external/
 CMD $python -m pip install --use-feature=2020-resolver --no-cache-dir -q -r requirements.txt --target=external/ --upgrade
 
-if [ -n "$optional" ]; then
-    CMD $python -m pip install --use-feature=2020-resolver --no-cache-dir -q $optional --target=external/ --upgrade
+if [ x"$1" == x"+docs" -o x"$2" == x"+docs" ]; then
+    CMD $python -m pip install --use-feature=2020-resolver --no-cache-dir -q -r docs/requirements.txt --target=external/ --upgrade
+    make -C docs PYTHON=$python
 fi
 
-if [ x"$1" == x"+docs" ]; then
-    CMD $python -m pip install --use-feature=2020-resolver --no-cache-dir -q -r docs/requirements.txt --target=external/ --upgrade
-    make -C docs
+if [ x"$1" == x"+pygelf" -o x"$2" == x"+pygelf" ]; then
+    CMD $python -m pip install --use-feature=2020-resolver --no-cache-dir -q pygelf --target=external/ --upgrade
```

Comment:
I'd get in a while loop where I'd be checking the value of `$1` in a `case` statement and act accordingly. At the end of the I'd `shift` the argument list. You need also `shift $OPTIND` before entering the loop. Check the `getopts` documentation in [bash(1)](https://linux.die.net/man/1/bash).